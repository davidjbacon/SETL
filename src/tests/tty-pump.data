-- SETL source for the subprogram invoked by tty-pump.setl
system ('tty >&2');  -- print name of stdin's pty on stderr
while (line := getline stdin) /= om loop
  print (hex line);
end loop;
printa (stderr, 'child exiting');
