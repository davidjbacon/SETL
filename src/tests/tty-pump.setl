-- The subprogram to open as a tty-pump is a SETL program in tty-pump.data
fd := open ('"$top_srcdir/run-setl" "$srcdir/tty-pump.data"', 'tty-pump');

-- Change to srcdir for convenience
chdir (getenv 'srcdir');

-- Use the source text of this program as test input
while (line := getline 'tty-pump.setl') /= om loop
  print (line);
  printa (fd, line);
  print (getline fd);
end loop;
printa (stderr, is_open 'tty-pump.setl');  -- should be #F
printa (stderr, is_open fd);  -- should be #T
close (fd);
printa (stderr, 'parent exiting');
