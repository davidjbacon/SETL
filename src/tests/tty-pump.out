-- The subprogram to open as a tty-pump is a SETL program in tty-pump.data
2D2D205468652073756270726F6772616D20746F206F70656E2061732061207474792D70756D702069732061205345544C2070726F6772616D20696E207474792D70756D702E64617461
fd := open ('"$top_srcdir/run-setl" "$srcdir/tty-pump.data"', 'tty-pump');
6664203A3D206F70656E2028272224746F705F7372636469722F72756E2D7365746C222022247372636469722F7474792D70756D702E6461746122272C20277474792D70756D7027293B


-- Change to srcdir for convenience
2D2D204368616E676520746F2073726364697220666F7220636F6E76656E69656E6365
chdir (getenv 'srcdir');
63686469722028676574656E76202773726364697227293B


-- Use the source text of this program as test input
2D2D205573652074686520736F757263652074657874206F6620746869732070726F6772616D206173207465737420696E707574
while (line := getline 'tty-pump.setl') /= om loop
7768696C6520286C696E65203A3D206765746C696E6520277474792D70756D702E7365746C2729202F3D206F6D206C6F6F70
  print (line);
20207072696E7420286C696E65293B
  printa (fd, line);
20207072696E7461202866642C206C696E65293B
  print (getline fd);
20207072696E7420286765746C696E65206664293B
end loop;
656E64206C6F6F703B
printa (stderr, is_open 'tty-pump.setl');  -- should be #F
7072696E746120287374646572722C2069735F6F70656E20277474792D70756D702E7365746C27293B20202D2D2073686F756C64206265202346
printa (stderr, is_open fd);  -- should be #T
7072696E746120287374646572722C2069735F6F70656E206664293B20202D2D2073686F756C64206265202354
close (fd);
636C6F736520286664293B
printa (stderr, 'parent exiting');
7072696E746120287374646572722C2027706172656E742065786974696E6727293B
